### PR TITLE
enigmail 2.0 support

### PR DESCRIPTION
**Pull request by shirosaki**
Fix for Thunderbird 60 and Enigmail 2.0.8.

- verify signed PGP MIME messages
- decrypt verify inline PGP messages

Quick reply with PGP doesn't work.

### CODE REVIEW

**Comment by Standard8** [comment]
Hi @shirosaki I've just merged a patch from the author of enigmail that fixes at least decrypting email, can you see if it fixes these other issues as well, or if we do need to do the more complex work here?

**Comment by shirosaki** [comment]
I've rebased on master.
I've fixed quick reply with PGP and inline PGP decryption bug.

**Comment by shirosaki** [comment]
@Standard8 
I'm using this patch and it works fine for me.
Could you review this?

File: modules/plugins/enigmail.js [hunk 1]
```diff
@@ -173,6 +179,16 @@ if (hasEnigmail) {
     global.EnigmailConstants = EnigmailCommon;
   }
 
+  let isEnigmail2 = false;
```
Comment (Standard8) [inline modules/plugins/enigmail.js]: I think we should drop pre-enigmail 2 support, but I'm happy not to do that in this PR, we can do a follow-up. I've filed #1305 to remind us.

File: modules/quoting.js [hunk 2]
```diff
@@ -146,8 +146,8 @@ function convertForwardedToBlockquote(aDoc) {
       let txt = child.textContent;
       let m = txt.match(re);
       if (child.nodeType == child.TEXT_NODE
-          && txt.includes("-----BEGIN PGP")
-          && txt.includes("----END PGP")
+          && !txt.includes("-----BEGIN PGP")
```
Comment (Standard8) [inline modules/quoting.js]: What is this change for?

Comment (shirosaki) [reply modules/quoting.js]: https://github.com/protz/thunderbird-conversations/commit/100641dbd663a3cebab55e279cc47e4078565821#diff-e54960bec9e47f840349eca45e3dcd94L171
Your commit changed that. It breaks inline pgp decription.
Not including is right.

Comment (Standard8) [reply modules/quoting.js]: Thank you for spotting that. Looks like I must have changed it manually and got it wrong :-(

I'll try and find time to look at the patch one more time in the next day or so.

File: modules/plugins/enigmail.js (hunk 1 above)
Comment (shirosaki) [reply modules/plugins/enigmail.js]: I agree.
There are a lot of code for old Enigmail.
I'll remove old code at the next time.

**Comment by Standard8** [comment]
Thanks for the patch. Sorry for the delay in getting to review this.

**Pull request merged**